lib/mapper.js

```diff
@@ -14,56 +14,31 @@ var each = require('@ndhoule/each');
 
 exports.track = function (msg, settings){
   var events = settings.events;
-  var conversionLabel = exports.getConversionLabel(msg.event(), events)
+  var whitelistedRemarketingEvent = exports.getRemarketingEvent(msg.event(), settings.whitelist);
+  var conversionLabel = exports.getConversionLabel(msg.event(), events);
   var remarketing = settings.remarketing;
   var ret = [];
 
   if (conversionLabel) {
-    var device = msg.proxy('context.device');
-    var app = msg.proxy('context.app');
-    var os = msg.proxy('context.os');
-    var referrer = msg.proxy('context.referrer.id');
-    var sdkVersion = app.name + '-sdk-' + firstLetterOfOS(device) + 'v' + app.build;
-    var idType = getIdType(device);
-    var limitedAdTracking = device.adTrackingEnabled ? 1 : 0;
-    var conversion = {
-      label: conversionLabel,
-      rdid: device.advertisingId,
-      idtype: idType,
-      lat: limitedAdTracking,
-      bundleid: app.namespace,
-      appversion: app.version,
-      osversion: os.version,
-      sdkversion: app.build,
-      value: msg.revenue() || 0,
-      currency_code: msg.currency() // defaults to 'USD'
-    };
-
-    if (referrer) conversion.referrer = referrer;
-
+    var conversion = formatConversionTag(msg);
+    conversion.label = conversionLabel;
     ret.push(conversion);
 
     // Just like client side, we only send remarketing tags for mapped events
     if (remarketing) {
-      var remarketingTag = {
-        bundleid: app.namespace,
-        rdid: device.advertisingId,
-        idtype: idType,
-        lat: limitedAdTracking,
-        remarketing_only: 1,
-        appversion: app.version,
-        osversion: os.version
-      };
-
-      // custom properties to remarket with must be prefixed with `data.` property
-      each(function(value, key) {
-        remarketingTag['data.' + key] = value;
-      }, msg.properties());
-
+      var remarketingTag = formatRemarketingTag(msg);
       ret.push(remarketingTag);
     }
   }
 
+  // You can still send remarketing pings by itself if you have whitelisted a certain event to do so
+  // Noop if we already sent the same remarketing tag above in case customer
+  // accidentally whitelists an event they already mapped as a conversion label
+  if (whitelistedRemarketingEvent && !remarketingTag) {
```

Comment:
I know its technically in scope, but could we move the definition of `remarketingTag` out so it looks like it should be in scope here?